systems/controllers/osc/operational_space_control.cc

```diff
@@ -146,28 +145,46 @@ void OperationalSpaceControl::AddAccelerationCost(
 }
 
 // Constraint methods
-void OperationalSpaceControl::AddContactPoint(const std::string& body_name,
-                                                 const VectorXd& pt_on_body) {
-  body_indices_.push_back(plant_wo_spr_.GetBodyByName(body_name).index());
-  pts_on_body_.push_back(pt_on_body);
+void OperationalSpaceControl::AddContactPoint(
+    const WorldPointEvaluator<double>* evaluator) {
+  single_contact_mode_ = true;
+  AddStateAndContactPoint(-1, evaluator);
 }
 
-void OperationalSpaceControl::AddStateAndContactPoint(int state,
-                                                         std::string body_name,
-                                                         VectorXd pt_on_body) {
-  fsm_state_when_active_.push_back(state);
-  AddContactPoint(body_name, pt_on_body);
+void OperationalSpaceControl::AddStateAndContactPoint(
+    int state, const WorldPointEvaluator<double>* evaluator) {
+  DRAKE_DEMAND(&evaluator->plant() == &plant_wo_spr_);
+
+  // Find the new contact in all_contacts_
+  auto it_c = std::find(all_contacts_.begin(), all_contacts_.end(), evaluator);
+  int contact_idx = std::distance(all_contacts_.begin(), it_c);
+  // Add to contact list if the new contact doesn't exist in the list
+  if (it_c == all_contacts_.cend()) {
+    all_contacts_.push_back(evaluator);
+  }
+
+  // Find the finite state machine state in contact_indices_map_
+  auto it_map = contact_indices_map_.find(state);
```

Comment:
I would recommend renaming it_map, it sounds like it's map of iterators when instead it's just an iterator

Reply:
I renamed it to `map_iterator`